src/packaging/requirements.py

```diff
@@ -56,8 +57,12 @@ def __init__(self, requirement_string: str) -> None:
             self.marker = Marker.__new__(Marker)
             self.marker._markers = _normalize_extra_values(parsed.marker)
 
-    def __str__(self) -> str:
-        parts: List[str] = [self.name]
+    @property
+    def canonical_name(self) -> str:
+        return canonicalize_name(self.name)
```

Comment:
Do we want to cache this? If not, this property seems redundant and perhaps we should call `canonicalize_name` directly instead.

Reply:
Done